### ISSUE TITLE
libevent erroneously detects kqueue bug on macOS 10.12

### ISSUE THREAD

**Issue by fanquake** [bitcoin/bitcoin#8818]
```
src/qt/bitcoin-qt
[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0
```

Seems to be this [bug](https://github.com/libevent/libevent/issues/376) in libevent. 

Opening an issue to track this as the warning could be a bit scary if it's ever output to users.

**Comment by laanwj** [comment]
At some point (don't know when, maybe even for 0.14) we probably want to build with the libevent 2.1.x series by default, which entered its beta phase start of 2015. This will make some other things possible in the official build, too:

- [WIP] rest: Stream entire utxo set #7759
- bitcoin-cli: More detailed error reporting #8722

**Comment by jonasschnelli** [comment]
@fanquake: is the brew version of libevent patched? Or is our build on OSX 10.12 broken (i'm still on 10.11).

**Comment by fanquake** [comment]
@jonasschnelli Looks like homebrew libevent stable (2.0.22) has been [patched](https://github.com/Homebrew/homebrew-core/pull/3349)

**Comment by RCasatta** [comment]
since I had 2.0.22 version of libevent I had to remove and reinstall it to include the patch cited by @fanquake 

`brew update
brew uninstall --ignore-dependencies libevent
brew install libevent`

I can confirm after the error isn't showed anymore

**Comment by fanquake** [comment]
Going to close this. 

Homebrew's libevent stable has been patched for this, and the 2.1.x branch doesn't have the issue (2.1.7-rc installed with `--devel`).

Also, now that libevent 2.1.7-rc is being used in depends, this should no-longer be a problem.
